lib/widget/switch.dart

```diff
@@ -50,28 +61,30 @@ class EnsembleSwitch extends StatefulWidget
     };
   }
 
-  void onToggle(bool newValue) {
-    setProperty('value', newValue);
+  void setSwitchValue(dynamic value) {
+    if (value is bool) {
+      _controller.value = Utils.getBool(value, fallback: false);
+    } else if (value is String) {
+      _controller.value = SwitchState.values.from(value) ?? _controller.value;
+    }
   }
-}
-
-class SwitchController extends FormFieldController {
-  bool value = false;
-  String? leadingText;
-  String? trailingText;
-  Color? activeColor;
-  Color? activeThumbColor;
-  Color? inactiveColor;
-  Color? inactiveThumbColor;
 
-  framework.EnsembleAction? onChange;
+  void onToggle(dynamic newValue) {
```

Comment:
same thing here. Each widget should have its own onToggle

Reply:
Updated it